Add section on keeping CI green (#1215)
python/devguide: getting-started/pull-request-lifecycle.rst

#. Comment on what is "good" about the pull request, not just the "bad". Doing
   so will make it easier for the PR author to find the good in your comments.

#. Look at any failures in CI on the current PR. See :ref:`"Keeping CI green"
   <keeping-ci-green>` below for simple things you can do to help move the PR forward.

Dismissing review from another core developer
---------------------------------------------

A core developer can dismiss another core developer's review if they confirmed
that the requested changes have been made.  When a core developer has assigned
the PR to themselves, then it is a sign that they are actively looking after
the PR, and their review should not be dismissed.

.. _keeping-ci-green:

> Keeping Continuous Integration green

Reviewer: ```suggestion
Keeping continuous integration green
```

Author: but that's CI, a proper name, like GitHub?

Reviewer: No, it's a common noun.

If short for continuous integration server/service, there are many named ones: Azure Pipelines, Jenkins, Gitlab Pipelines, AppVeyor. The proper noun of a CI service we're using is GitHub Actions.

If continuous integration as the name for a practice, it's also a common noun. For example, see https://en.wikipedia.org/wiki/Continuous_integration

====================================

Our change management workflows generally won't allow merging PRs with

> failures on them. Therefore, if you see a CI failure on a PR, have a look

Reviewer: ```suggestion
failures. Therefore, if you see a CI failure on a PR, have a look
```

what it is about.

Usually the failure will be directly related to the changes in the current
PR. If you happen to have any insight into the failure, let the author know
in a review comment. CI runs sometimes generate thousands of lines of output.
Even something as simple as finding the traceback and putting it in the
comment will be helpful to the PR author.

If the failure doesn't look related to the change you're looking at, check
if it's not present on the `Release Status`_ Buildbot dashboard as well.
If so, that means the failure was introduced in a prior change. Using Buildbot's

> UI you can find which PR introduced the issue and comment there that it

Reviewer: ```suggestion
UI you can find which PR introduced the issue and comment that it
```

affects other PRs.

If you still don't see where the failure originates from, check for

> an "This branch is out-of-date with the base branch" sign next to the

Reviewer 2: ```suggestion
a "This branch is out-of-date with the base branch" sign next to the
```

list of executed checks. Clicking "Update branch" next to this message
will merge in the latest changes from the base branch into the PR.

If this still doesn't help with the failure on the PR, you can try
to re-run that particular failed check. Go to the red GitHub Action job,

> and find a button on the top right called "Re-run jobs". Select

Reviewer 3: ```suggestion
click on the "Re-run jobs" button on the top right, and select
```

"Re-run failed jobs". The button will only be present when all other jobs

> finished running.

Reviewer: Only applies to core devs and triagers?

Reviewer 2: That's also true for the previous paragraph (about merging in main).

Reviewer: Any contributor can merge `main` into their PR, for example: https://github.com/python/cpython/pull/105984#commits-pushed-3cd8d82
